tuning/scripts/stage_tuning.sh

```diff
@@ -1,86 +1,87 @@
 #!/bin/bash
 
-
-make_tensile_tuning () {
-
-    local FULLFILENAME=$1
-
-    local BUILDNAME="build"
-
-    local FILENAME=$(basename "$FULLFILENAME")
-    local FNAME="${FILENAME%.*}"
-    local FILEPATH=$(ls $SOURCE/$FILENAME)
-
-    local WORKINGPATHNAME=${BUILDNAME}-${FNAME}
-    local WORKINGPATH=${DESTINATION}/${WORKINGPATHNAME}
-
-    mkdir -p $WORKINGPATH
-    cp $FILEPATH $WORKINGPATH
-    pushd ${WORKINGPATH} > /dev/null
-    echo "#!/bin/sh" > doit.sh
-    echo "if [ ! -d 3_LibraryLogic ] || [ \$(ls -A 3_LibraryLogic | wc -c) -eq 0 ]; then" >> doit.sh
-    echo "  touch time.begin" >> doit.sh
-    echo "  ../Tensile/bin/Tensile $FILENAME ./ > make.out 2>&1" >> doit.sh
-    echo "  touch time.end" >> doit.sh
-    echo "fi" >> doit.sh
-
-    chmod +x doit.sh
-    popd > /dev/null
+function make_tensile_tuning() {
+
+  local FILE_PATH=$1
+
+  local FILE_NAME; FILE_NAME=$(basename "${FILE_PATH}")
+  local FILE_PATH; FILE_PATH=$(ls "$SOURCE/$FILE_NAME")
+  local FILE_NAME_NO_EXT="${FILE_NAME%.*}"
+
+  local WORKING_PATH="${DESTINATION}/build-${FILE_NAME_NO_EXT}"
+
+  mkdir -p "${WORKING_PATH}"
+  cp "${FILE_PATH}" "${WORKING_PATH}"
+  pushd "${WORKING_PATH}" > /dev/null || exit
+  {
+    echo "#!/bin/sh"
+    echo "if [ ! -d 3_LibraryLogic ] || [ -z \"\$(ls -A 3_LibraryLogic)\" ]; then"
+    echo "  touch time.begin"
+    echo "  ${TENSILE}/Tensile/bin/Tensile ${FILE_NAME} ./ > make.out 2>&1"
+    echo "  touch time.end"
+    echo "fi"
+  } > doit.sh
+
+  chmod +x doit.sh
+  popd > /dev/null || exit
```

Comment:
"doit.sh" for me doesn't quite "do it" heh. I am hoping to rename this to something more descriptive, say launchTensileBenchmark.sh or something better.

Reply:
Yah, this was my half baked naming convention from my original version of the automation. It should be changed to something meaningful.